mne/preprocessing/xdawn.py

```diff
@@ -53,6 +55,12 @@ def _least_square_evoked(epochs_data, events, tmin, sfreq):
         Start time before event.
     sfreq : float
         Sampling frequency.
+    solver : str | function
+        Either a function which takes the sparse predictor matrix X and the
+        observation matrix Y, and returns the coefficient matrix b;
+        or a string. If str, must be ``'cholesky'``, in which case the solver
+        used is ``linalg.solve(dot(X.T, X), dot(X.T, y))``, or ``'pinv'``,
+        in which case a solver based on a pseudo-inverse is used.
```

Comment:
`pinv` should probably not be the default. IIRC it is not as numerically stable or fast as other solvers. Maybe `linalg.lstsq`?